backends/dlc/dlcmanager.h

```diff
@@ -0,0 +1,94 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef DLC_DLCMANAGER_H
+#define DLC_DLCMANAGER_H
+
+#include "common/str.h"
+#include "common/str-array.h"
+#include "common/queue.h"
+#include "common/singleton.h"
+#include "backends/dlc/store.h"
+
+namespace DLC {
+
+enum State {
+	kAvailable,
+	kInProgress,
+	kDownloaded,
+	kCancelled
+};
+
+struct DLC {
+	Common::String name;
+	Common::String id;
+	uint32 size;
+	uint32 idx;
+	State state;
+};
+
+class DLCManager : public Common::Singleton<DLCManager> {
+
+	Common::Array<DLC*> _dlcs;
+	Common::Queue<DLC*> _queuedDownloadTasks;
+
+	Store* _store;
+
+	bool _isDLCDownloading = false;
+	Common::String _currentDownloadingDLC;
+
+public:
+	DLCManager();
+	virtual ~DLCManager() {}
+
+	void init();
+
+	// Runs only once in init()
+	void getAllDLCs(Common::Array<DLC*>& dlcs);
+
+	// Requested by GUI to show all available DLCs in simple list view
+	Common::U32StringArray getDLCList();
```

Comment:
can maybe be `const Common::U32StringArray& getDLCList() const;` to prevent the copy